src/Contract/ClaimAuctionLotBidder.purs

```diff
@@ -0,0 +1,289 @@
+  , AuctionEscrowState(BiddingStarted, AuctionConcluded)
+  , AuctionInfo(AuctionInfo)
+  , AuctionInfoValidationError
+  , AuctionTerms(AuctionTerms)
+  , AuctionTermsValidationError
+  , ContractOutput
+  , ContractResult
+  , StandingBidRedeemer(ConcludeAuctionRedeemer)
+  , StandingBidState
+  , Utxo
+  , emptySubmitTxData
+  , mkContractOutput
+  , sellerPayout
+  , submitTxReturningContractResult
+  , totalAuctionFees
+  , validateAuctionInfo
+  , validateAuctionTerms
+  )
+import HydraAuctionOffchain.Contract.Validators (MkAuctionValidatorsError, mkAuctionValidators)
+import HydraAuctionOffchain.Helpers (getInlineDatum)
+
+claimAuctionLotBidderContract :: AuctionInfo -> Contract (ContractOutput TransactionHash)
+claimAuctionLotBidderContract =
+  mkContractOutput _.txHash <<< mkClaimAuctionLotBidderContractWithErrors
+
+mkClaimAuctionLotBidderContractWithErrors
+  :: AuctionInfo
+  -> ExceptT ClaimAuctionLotBidderContractError Contract ContractResult
+mkClaimAuctionLotBidderContractWithErrors auctionInfo = do
+  let
+    AuctionInfo auctionInfoRec = auctionInfo
+    auctionCs = auctionInfoRec.auctionId
+    auctionTerms@(AuctionTerms auctionTermsRec) = auctionInfoRec.auctionTerms
+
+  -- Check auction terms:
+  validateAuctionTerms auctionTerms #
+    validation (throwError <<< ClaimAuctionLotBidder_Error_InvalidAuctionTerms) pure
+
+  -- Check that the current time is within the purchase period:
+  nowTime <- lift currentTime
+  when (nowTime < auctionTermsRec.biddingEnd) $
+    throwError ClaimAuctionLotBidder_Error_CurrentTimeBeforeBiddingEnd
+  when (nowTime >= auctionTermsRec.purchaseDeadline) $
+    throwError ClaimAuctionLotBidder_Error_CurrentTimeAfterPurchaseDeadline
+
+  -- Build validators:
+  validators <-
+    withExceptT ClaimAuctionLotBidder_Error_CouldNotBuildAuctionValidators $
+      mkAuctionValidators auctionCs auctionTerms
+
+  -- Check auction info:
+  validateAuctionInfo auctionInfo validators #
+    validation (throwError <<< ClaimAuctionLotBidder_Error_InvalidAuctionInfo) pure
+
+  -- Query current auction escrow utxo:
+  auctionEscrowUtxo <- queryAuctionEscrowUtxo BiddingStarted auctionInfo
+    !? ClaimAuctionLotBidder_Error_CouldNotFindAuctionEscrowUtxo
+
+  -- Query standing bid utxo:
+  standingBidUtxo /\ standingBid <- queryStandingBidUtxo auctionInfo
+    !? ClaimAuctionLotBidder_Error_CouldNotFindStandingBidUtxo
+
+  -- Get bid terms:
+  bidTerms <- unwrap standingBid ?? ClaimAuctionLotBidder_Error_EmptyStandingBid
+
+  -- Get buyer pkh:
+  buyerPkh <- wrap <$> toPubKeyHash (unwrap (unwrap bidTerms).bidder).bidderAddress
+    ?? ClaimAuctionLotBidder_Error_CouldNotGetBuyerPkh
+
+  -- Get seller pkh:
+  sellerPkh <- wrap <$> toPubKeyHash auctionTermsRec.sellerAddress
+    ?? ClaimAuctionLotBidder_Error_CouldNotGetSellerPkh
+
+  let
+    validatorHashes = unwrap $ validatorHash <$> validators
+
+    mkAuctionToken :: TokenName -> Value
+    mkAuctionToken tokenName = Value.singleton auctionCs tokenName one
```

Comment:
Don't we have something like `foo = \cs tn -> Value.singleton cs tn one`?

Reply:
I think using `Value.singleton` is the "canonical" way to build a single-token Value.